### PR TITLE
Add configuration for kv options

### PR DESCRIPTION
**Pull request by cw9**
@xichen2020 @jeromefroe

### CODE REVIEW

File: kv/config.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```
Comment (jeromefroe) [inline kv/config.go]: nit: 2017?

File: kv/config.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package kv
+
+// Configuration configs the kv options.
```
Comment (jeromefroe) [inline kv/config.go]: nit: s/configs the/is the config for

File: kv/config.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+type Configuration struct {
+	Environment string `yaml:"environment" validate:"nonzero"`
+	Namespace   string `yaml:"namespace" validate:"nonzero"`
```
Comment (xichen2020) [inline kv/config.go]: Does the config always require a non-empty namespace?

File: kv/config.go (hunk 1 above)
Comment (cw9) [reply kv/config.go]: sg

File: kv/config.go (hunk 2 above)
Comment (cw9) [reply kv/config.go]: sg

File: kv/config.go (hunk 3 above)
Comment (cw9) [reply kv/config.go]: yeah, if you override, you'd better provide override for both fields

**Pull request merged**